code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -268,3 +270,88 @@
 /mob/living/simple_animal/hostile/construct/harvester/hostile //actually hostile, will move around, hit things
 	AIStatus = AI_ON
 	environment_smash = 1 //only token destruction, don't smash the cult wall NO STOP
+
+
+////////////////Glow////////////////////
+/mob/living/simple_animal/hostile/construct/proc/updateglow()
+	overlays = 0
+	var/overlay_layer = LIGHTING_LAYER + 1
+	if(layer != MOB_LAYER)
+		overlay_layer=TURF_LAYER+0.2
+
+	overlays += image(icon,"glow-[icon_state]",overlay_layer)
+	set_light(1, 1, l_color = "#FFFFFF")
+
+
+///ui stuff
+
+/mob/living/simple_animal/hostile/construct/armoured/handle_hud_icons_health()

```

Comment:
okay for these you want to do, like, this, with [construct_health_string_var] being a var that determines what icon it needs, instead of having one for each type of construct.

    /mob/living/simple_animal/hostile/construct/update_health_hud()
	if(!client || !hud_used)
		return
	if(hud_used.healths)
		if(stat != DEAD)
			if(health >= maxHealth)
				hud_used.healths.icon_state = "[construct_health_string_var]_health0"
			else if(health > maxHealth*0.8)
				hud_used.healths.icon_state = "[construct_health_string_var]_health2"
			else if(health > maxHealth*0.6)
				hud_used.healths.icon_state = "[construct_health_string_var]_health3"
			else if(health > maxHealth*0.4)
				hud_used.healths.icon_state = "[construct_health_string_var]_health4"
			else if(health > maxHealth*0.2)
				hud_used.healths.icon_state = "[construct_health_string_var]_health5"
			else
				hud_used.healths.icon_state = "[construct_health_string_var]_health6"
		else
			hud_used.healths.icon_state = "[construct_health_string_var]_health7"